Client/mods/deathmatch/logic/luadefs/CLuaEngineDefs.cpp

```diff
@@ -2560,3 +2591,188 @@ void CLuaEngineDefs::EnginePreloadWorldArea(CVector position, std::optional<Prel
     if (option == PreloadAreaOption::ALL || option == PreloadAreaOption::COLLISIONS)
         g_pGame->GetStreaming()->LoadSceneCollision(&position);
 }
+
+std::variant<bool, CClient2DFX*> CLuaEngineDefs::EngineAddModel2DFX(lua_State* luaVM, std::uint32_t modelID, CVector position, e2dEffectType effectType, std::unordered_map<std::string, std::variant<bool, float, std::string>> effectData)
+{
+    // Only these effects make sense in MTA
+    if (effectType != e2dEffectType::LIGHT && effectType != e2dEffectType::PARTICLE && effectType != e2dEffectType::ROADSIGN &&
+        effectType != e2dEffectType::ESCALATOR && effectType != e2dEffectType::SUN_GLARE)
+        return false;
+
+    if (!CClient2DFXManager::IsValidModel(modelID))
+        throw std::invalid_argument("Invalid model ID");
+
+    const char* error = CClient2DFXManager::IsValidEffectData(effectType, effectData);
+    if (error)
+        throw std::invalid_argument(error);
```

Comment:
Its better to throw LuaFunctionError instead of invalid argument if its runtime error